### DISCUSSION TITLE
Examples for creating a sensor that watches for GCP pub/sub messages

### DISCUSSION THREAD

**Discussion by gnilrets** [dagster-io/dagster#19615] [Q&A]
Anyone have an example of using a sensor to trigger actions based on a pub/sub event?

**Comment by joscha** [comment]
@gnilrets did you ever find one?

**Reply by gnilrets** [reply]
I ended up creating a sensor that polls for new messages in pub sub every 30 seconds.  It seems to work ok, but I don't think it would scale well if we start getting tons of messages as it seems to go through fits every few weeks where it will fail due to timeouts 4-5 times over a few hours and then be quiet for a few more weeks.

**Comment by joscha** [comment]
references https://github.com/dagster-io/dagster/issues/2415
